### DISCUSSION TITLE
Updating Filecoin’s value accrual mechanisms

### DISCUSSION THREAD

**Discussion by AxCortes** [filecoin-project/FIPs#1010] [Enhancements - Technical]
**Background**

One important goal in Filecoin is to drive real user adoption. Specifically we want to onboard more real user paid data deals. There are, however, no direct mechanisms in place through which real user adoption would lead to FIL value accrual.

The only way value accrues onto Filecoin as user demand rises is through base fee burn. This is appropriate for other blockchains whose main product users consume is their block space, but it is not appropriate for Filecoin where the main product offered is provable data storage. This issue has been previously presented at length https://github.com/filecoin-project/FIPs/discussions/557, where a suggested solution is to introduce explicit proof fees, where value would accrue to Filecoin every time someone uses a storage proof, regardless of the amount of gas the proof consumes.

Recently a lot of discussion has begun on the possibility of introducing new types of proofs, enabling different types of storage, such as Proof of Data Possession (PDP) enabling Filecoin hot storage. The value of these proofs to the users would be even more disconnected from their respective gas usage, since these proofs by necessity must be less gas-expensive, while providing more value to the users, than cold storage proofs.

A separate issue is that, while we expect a rise in paid deals would lead to a rise in demand for FIL (to be used as payment for such deals), there are no actual incentives that make users prefer to pay for their deals in FIL. In fact large part of deal payments may happen completely off chain.

**Proposed solutions**

1. Explicit proof fees

While initial discussion on explicit proof fees lost some momentum following their introduction in https://github.com/filecoin-project/FIPs/discussions/557, we propose actively restarting this line, specially in light of the hot storage and PDP/ new proofs discussion.  As we diversify the products and services Filecoin offers, the disconnect between the value they provides to users, and the base fee burn will grow. 

In particular, if new hot storage capabilities as are being discussed in https://github.com/filecoin-project/FIPs/discussions/988 are introduced, these should be accompanied with appropriate proof fees, such that wide adoption of Filecoin hot storage actually leads to value accrual to Filecoin. 

As was noted in the original proof fees proposal, having appropriate proof fees, would also enable us to remove the controversial Batch Balancer mechanism, which aims to address similar issues, but has significant flaws.

2. Incentivizing for native paid deals

We can explore mechanisms to incentivize deal payments to actually happen in FIL and on chain, such that increased demand for storage translates to increased demand for FIL. One direct mechanism is to award sector quality multipliers based on amount of FIL deal payment shown in the sector. This could be organized such that it increases competition amongst sectors to include larger FIL payments on chain.

**Comment by mjroddy** [comment]
Thanks @AxCortes for the write up.

Quick question I had on initial pass:

1. The idea you're proposing is that an SP should incur higher fees for hot storage beyond what is technically required because it adds more value to the client? Or are you talking about just having seperate fee markets for different proofs?

2. Less of a question here. Whilst I understand the want by holders (and economy designers) to use FIL within the filecoin economy why would an SP want to take on more FIL exposure, their costs remain in USD. The general feedback I get is SP's wanting a stable revenue stream along side the FIL.

But regardless of what the SP wants I think this just isn't what clients want. Have we had clients request this at all? If that's not the case why would we build something to incentivise this. I think better time could be spent incentivising other activity on the network that spurs adoption opposed to incentivising a payment mechanism that's going to cause clients/SP's problems. Not saying we can't accept FIl but do we need the incentive for the clients. Defi could surely help in enabling FIL/USD hedging or payment deals.

Am I missing something there?

**Reply by AxCortes** [reply]
> The idea you're proposing is that an SP should incur higher fees for hot storage beyond what is technically required because it adds more value to the client? Or are you talking about just having seperate fee markets for different proofs?

I'm saying something like new proposed hot storage proofs, kind of break the way in which adoption by users translates into value accrual to Filecoin. The current system sort of works, only because current proofs (like PoRep) are actually quite gas intensive. So higher user demand, means proportionally more demand for gas usage, which leads to deflationary pressure through higher burn. In #557 they explain in more length why its not a great idea to use gas usage as a proxy for the service we are actually providing. But what I am adding here is that this argument is amplified in the case of hot storage. For hot storage proofs, gas usage would be an even worse proxy for value provided than it is now for things like PoRep.

> why would an SP want to take on more FIL exposure

That's pretty much the point, that they wouldn't. So even if Filecoin as a data storage service becomes very successful, and many users want to store their data on Filecoin, it doesn't mean any of the deals will actually happen in FIL, or create any demand for using FIL. The question is should align incentives such that it is actually favorable to the SP to have the deal made in FIL?

I'm not proposing that deals have to be made on FIL, there is still choice, but can we make it more favorable to actually use FIL?

**Reply by mjroddy** [reply]
Ok I understand the issue better, will need to think through this more.

I strongly disagree with the notion of incentivising FIL deals for the sake of it. I don't think it actually even captures that much value. Client A buys FIL, pays for some storage to SP B who then sells to pay bills. It's more of a velocity problem.

I think we could come up with better ways of Filecoin as a network capturing value than just enforcing a payment mechanism between client and SP. I.e users wanting to store data burn FIL (ideally the user not even aware they are using FIL) and SP's storing the data earn minted FIL. With the idea of burn>mint equals price adjustment up for holders.

Maybe your post was just to start the discussion which I hope it does but not the way I would go about it.

**Reply by AxCortes** [reply]
> I.e users wanting to store data burn FIL (ideally the user not even aware they are using FIL) and SP's storing the data earn minted FIL.

I think this is exactly the idea of proof fees. User wanting to store some data, means a proof enabled by Filecoin was used, and this created value. Ideally this is a cost that either the user pays, or the SP passes on to the user indirectly.


> Client A buys FIL, pays for some storage to SP B who then sells to pay bills. It's more of a velocity problem.

Yes I agree this quick buying and selling of FIL doesn't do that much, perhaps if the deal payment is vested over time, which is what a user ideally would want, then there is some locking FIL effect.

> Maybe your post was just to start the discussion which I hope it does but not the way I would go about it.

For sure, I have my own thoughts, but this is more about the discussion and seeing what everyone else thinks about the issue.

**Reply by mjroddy** [reply]
So I think the only way that value accrual occurs (with current design) is through burning. Proof fees do make sense and the SP can cover the cost as it's a value add to user (I'm still understanding whether that is artificially higher for the sake of it or just another fee market).

I'm sure that something could be created where by the client pays in USDC or similar, it's converted to FIL for the deal (vested as you say) and then options to hedge out the volatility if either the client/SP wants that but really that's pretty minimal in terms of effect imo and again not a great UX. incentivising FIL deals isn't creating value for clients.

We have to remove FIL from circulation permanently, today that's occurring through burn. I think it would be better to focus on that, either through proofs for valued services, or transaction volume from applications that are providing value to their users.

**Reply by nicola** [reply]
I actually totally agree with your question - I think the ideal scenario is when deals are paid in any currency (bigger markets) and the Filecoin network takes a fee (converts say USDC into FIL and burns it)

**Reply by AxCortes** [reply]
> I actually totally agree with your question - I think the ideal scenario is when deals are paid in any currency (bigger markets) and the Filecoin network takes a fee (converts say USDC into FIL and burns it)

I think you are just describing Explicit proof fees, which I agree with too, and are the first point of the proposal, so agree we should definitely do that.

It is also not mutually exclusive with the idea of also locking FIL for deals, it's a separate approach we can explore.  I agree burning does more than locking for value accrual, but the effect of locking should not be dismissed either.

**Reply by anorth** [reply]
@AxCortes I think the resistance to part 2 of incentivising paid deals to use FIL is that it would seem to degrade the product offering and also probably not work. SPs can offer a better service to clients by trading in the tokens that are convenient for the client. If SPs are paid through the back door to offer a less convenient service, they'll get less business and it would seem that the network as a whole is providing less value (leaving less for FIL to capture too). As @mjroddy points out, restrictions or incentives would likely be worked around, hedged, intermediated etc to provide the service that the parties want anyway. Restrictions to thwart such arrangements (like requiring more locking) just further degrade the service offering.

> So even if Filecoin as a data storage service becomes very successful, and many users want to store their data on Filecoin, it doesn't mean any of the deals will actually happen in FIL, or create any demand for using FIL. The question is should align incentives such that it is actually favorable to the SP to have the deal made in FIL?

The problem is well stated, but I take the opposite strategic direction from it. The best service with many users doesn't necessitate any use of FIL for medium-of-exchange, so how can we design Filecoin to thrive and capture a fair share of value despite this. Take it as a design axiom that we can't force use of FIL for deals, and design an economically sustainable system from there.

I think part 1 of onboarding/service fees is a good, probably sufficient solution, so long as the network can effectively charge a fee associated with the service value (e.g. fee market).

**Reply by anorth** [reply]
> an SP should incur higher fees for hot storage beyond what is technically required because it adds more value to the client? Or are you talking about just having seperate fee markets for different proofs?
> ... I'm still understanding whether that is artificially higher for the sake of it or just another fee market

@mjroddy Here's how I think about these things. What is the service that the Filecoin network provides? 
- It is not providing storage. SPs provide storage, but "Filecoin" doesn't.
- It is not performing proofs. The SP does that too.
- It's only kind-of checking proofs.

What the Filecoin network provides is a coordination mechanism that helps clients and providers make a transaction. It establishes a standard protocol for creating and checking proofs, and pays thousands of computers around the world to check those proofs. It provides a tightly integrated programmable system for coding rules about service faults, payments etc. (It also provides some basic economic blocks like pledge and penalties, though arguably some of these built-in ones are not of value to the participants). This coordination system provides a clear baseline set of rules and a basis on which clients can _trust_ that their data is replicated. Filecoin provides this coordination and trustworthy system to SPs, who can then use it to sell verifiable data storage services to clients.

A client and SP could do all this themselves, without a blockchain. An SP could seal sectors and send proofs, including daily Window PoST, directly to the client, which could check them. This would be a bit of a nightmare to arrange, multiplied by many clients-provider pairs, and a huge barrier to use. The key value that Filecoin provides is one network that does this for everyone. And the more clients and SPs are using the same network, the better it gets for everyone: more secure, more technology and product development, etc etc.

Imagine for a moment that blockchain capacity was unbounded and so gas was free (but many validators still secured the network for some reason). In this case we might say that no fee is _technically_ required. But Filecoin should still charge a fee for the service that it provides to the SPs by practically enabling this verifiable storage. The fee can't be so high that it raises the cost to clients too much, but it can certainly be above the pure technology costs.

**Reply by mjroddy** [reply]
@anorth taking this a step further, should (is it possible) the filecoin network charge for these services in a flat USD figure, payable in FIL.

It could burn the entire fee or cycle some back to a development fund for grants and funding etc.

That means that even if gas was free SP's would still pay for the proof check to provide assurance about their own services to clients 

Zooming out the filecoin network becomes deflationary when the value of services to clients (measured through fees) is greater than the incentive for them to join....

**Comment by Fatman13** [comment]
> One direct mechanism is to award sector quality multipliers based on amount of FIL deal payment shown in the sector. This could be organized such that it increases competition amongst sectors to include larger FIL payments on chain.

How can protocol tell if a deal payment is genuine as opposed to miners spamming to get quality multipliers?

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
FIL+

**Reply by AxCortes** [reply]
It doesn't, but the point is a non-genuine payment has the same cost, the SP has to lock up the amount of FIL in the deal that otherwise a real client would have.  

So there are not really "fake deals", it is a real deal, where the SP is the paying client, which they are free to do.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
FIL+

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
we could just do PoS if locking up FIL is the value we are out for

**Reply by AxCortes** [reply]
I agree the idea is similar to just including an additional staking element to get more rewards. One point of the design should be that the additional rewards wouldn't be enough for the SP to just go and lock their own FIL. The finances would make sense if it is the client's payment. Suppose a client wanted to pay an SP and spread the payment equally throughout the Sector lifetime, They are already locking some amount of funds with a vesting schedule. We'd just want to make it favorable for those funds to be FIL, but not necessarily incentivice locking funds that were not going to be locked anyway.

**Reply by Fatman13** [reply]
> It doesn't, but the point is a non-genuine payment has the same cost, the SP has to lock up the amount of FIL in the deal that otherwise a real client would have.

i might be missing something and don't quite get it. It says...

> award sector quality multipliers based on amount of FIL deal payment shown in the sector

can't one just make fake deal payment to get multipliers? like paying back client the deal payment amount?
